components/textbook-beta/DynamicBackgroundLogo.vue

```diff
@@ -0,0 +1,251 @@
+      this.animationProgress = newAnimationProgress
+      this.renderingContext!.clearRect(0, 0, this.canvas!.width, this.canvas!.height)
+
+      for (let i = 0; i < this.maskWidth; i++) {
+        for (let j = 0; j < this.maskHeight; j++) {
+          const gridElement = this.gridData[j][i]
+          if (gridElement.value) {
+            const alpha = Math.pow(this.clamp(this.animationProgress * 2 - gridElement.delay), 2) * 2
+            this.renderingContext!.strokeStyle = `rgba(255, 255, 255, ${alpha})`
+            const finalSize = this.squareSize / gridElement.value
+            this.renderingContext!.strokeRect(SPACING_BETWEEN_CELLS + i * this.squareSizeWithSpacing, SPACING_BETWEEN_CELLS + j * this.squareSizeWithSpacing, finalSize, finalSize)
+          }
+        }
+      }
+    },
+    resize () {
+      this.parentWidth = this.parent!.clientWidth
+      this.parentHeight = this.parent!.clientHeight
+      this.squareSizeWithSpacing = (this.parentWidth - 2 * SPACING_BETWEEN_CELLS) / Math.max(this.maskHeight, this.maskWidth)
+      this.squareSize = this.squareSizeWithSpacing - SPACING_BETWEEN_CELLS
+
+      const pixelDensityConsideringDevicePixelRatio = (window.devicePixelRatio || 1) * this.pixelDensity
+
+      // this disparity of html-width/height and css-width/height
+      // creates a high resolution canvas with better lines
+      this.canvas!.width = this.parentWidth * pixelDensityConsideringDevicePixelRatio
+      this.canvas!.height = this.parentHeight * pixelDensityConsideringDevicePixelRatio
+      this.canvas!.style.width = `${this.parentWidth}px`
+      this.canvas!.style.height = `${this.parentHeight}px`
+
+      this.renderingContext!.scale(pixelDensityConsideringDevicePixelRatio, pixelDensityConsideringDevicePixelRatio)
+      this.animationProgress = Math.min(0.9, this.animationProgress)
+    },
+    // randomize mask with the quantum probabilities
+    randomizeMask () {
+      this.gridData = imgMask.map(row => row.map((maskValue: number) : GridCell => {
+        const randomSuccess = Math.random()
+        const randomDelay = Math.random()
+        const success1 = randomSuccess <= SUCCESS_RATIO_1
+        const success2 = randomSuccess > SUCCESS_RATIO_1 && randomSuccess <= ACCUMULATED_SUCCESS_RATIO
+
+        if (maskValue === 1) { // 1 accept both states
+          return { value: success1 ? 1 : success2 ? 2 : 0, delay: randomDelay }
+        }
+
+        if (maskValue === 2) { // 2 accept only second state
+          return { value: success2 ? 2 : 0, delay: randomDelay }
+        }
+
+        return { value: 0, delay: 0 }
+      }))
+    },
+    clamp (value: number) : number {
+      return Math.min(Math.max(value, 0), 1)
+    },
+    interactionRedraw () {
+      this.animationProgress = 0.5
+      this.drawSpeed = this.initialDrawSpeed * DRAWSPEED_CORRECTION * 3
+      this.randomizeMask()
+    }
+  }
+})
+</script>
+
+<style lang="scss" scoped>
+.dynamic-background-logo {
+  &-background-canvas {
```

Comment:
Small change follow our BEM convention:

```suggestion
  &__background-canvas {
```